### PR TITLE
 Bug Fixed : Matrix power of identity matrix fails #9823

### PR DESCRIPTION
**Pull request by kevalds51**
Bug Fixed : Matrix power of identity matrix fails #9823

### CODE REVIEW

File: sympy/matrices/expressions/matexpr.py [hunk 1]
```diff
@@ -116,6 +116,8 @@ def __pow__(self, other):
             return Identity(self.rows)
         elif other is S.One:
             return self
+        elif self.is_Identity == True:
```
Comment (moorepants) [inline sympy/matrices/expressions/matexpr.py]: `elif self.is_Identity

**Comment by moorepants** [comment]
Please add a unit test that exercises this code change.

**Comment by kevalds51** [comment]
My bad. 
Did some other important changes too that i missed earlier. 
I will close this request and start a new one

**Comment by moorepants** [comment]
There is no need to close this request. Just push your changes to the same branch and this PR will update.

**Comment by kevalds51** [comment]
Ok. New to GIT, but I get it now.

**Pull request closed without merging**